### PR TITLE
Fix shellcheck failures in run_gluster.sh,create_block.sh,run_nfs.sh

### PR DESCRIPTION
**Pull request by xichengliudui**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by xichengliudui** [comment]
/assign @BenTheElder 
Ref: https://github.com/kubernetes/kubernetes/pull/76123

File: test/images/volume/gluster/run_gluster.sh [hunk 1]
```diff
@@ -14,24 +14,24 @@
...
 {
-    mount -t tmpfs test $DIR
-    chmod 755 $DIR
-    cp /vol/* $DIR/
+    mount -t tmpfs test "$DIR"
+    chmod 755 "$DIR"
+    cp /vol/* "$DIR/"
     /usr/sbin/glusterd -p /run/glusterd.pid
-    gluster volume create test_vol `hostname -i`:$DIR force
+    gluster volume create test_vol "$(hostname -i)":"$DIR" force
```
Comment (BenTheElder) [inline test/images/volume/gluster/run_gluster.sh]: nit:
```suggestion
    gluster volume create test_vol "$(hostname -i):${DIR}" force
```

File: test/images/volume/iscsi/create_block.sh [hunk 2]
```diff
@@ -39,9 +39,9 @@ dd if=/dev/zero of=block seek=120 count=1 bs=1M
 mkfs.ext2 block
 
 # Add index.html to it
-mount -o loop block $MNTDIR
-echo "Hello from iSCSI" > $MNTDIR/index.html
-umount $MNTDIR
+mount -o loop block "$MNTDIR"
+echo "Hello from iSCSI" > "$MNTDIR/index.html"
```
Comment (BenTheElder) [inline test/images/volume/iscsi/create_block.sh]: nit:
```suggestion
echo "Hello from iSCSI" > "${MNTDIR}/index.html"
```
when we're mixing variables and literals we should use braces around the variables

Comment (xichengliudui) [reply test/images/volume/iscsi/create_block.sh]: YES, thanks.

**Comment by xichengliudui** [comment]
has been updated.

File: test/images/volume/nfs/run_nfs.sh [hunk 3]
```diff
@@ -22,21 +22,23 @@ function start()
     while getopts "G:" opt; do
         case ${opt} in
             G) gid=${OPTARG};;
+            *) echo "usage: $0 [-v] [-r]" >&2
```
Comment (BenTheElder) [inline test/images/volume/nfs/run_nfs.sh]: isn't the only accepted option -G?

**Comment by xichengliudui** [comment]
sorry, i need rebase.

File: test/images/volume/nfs/run_nfs.sh (hunk 3 above)
Comment (xichengliudui) [reply test/images/volume/nfs/run_nfs.sh]: change to: `G) echo "usage: $0 [-v] [-r]" >&2` ?

Comment (cblecker) [reply test/images/volume/nfs/run_nfs.sh]: why -v -r? Those aren't the possible usage options.

Comment (xichengliudui) [reply test/images/volume/nfs/run_nfs.sh]: I am not very familiar with this document, may I ask if you have any good Suggestions for modification?

Comment (cblecker) [reply test/images/volume/nfs/run_nfs.sh]: I think `[-G]` is all you need, as that's the only possible option.

Comment (xichengliudui) [reply test/images/volume/nfs/run_nfs.sh]: Thank you both. Is that right?

**Comment by xichengliudui** [comment]
Has been merged

**Pull request closed without merging**